### PR TITLE
cmd/juju/user: add "logout" command

### PR DESCRIPTION
**Pull request by axw**
"juju logout" will remove the user credentials
for the specified/current controller. If there
is a password but no macaroon recorded for the
user, then the user must either acquire a macaroon
by changing the password first, or use the --force
flag.

We now disallow a client logging in as multiple
users for a controller simultaneously, and
consequently drop the "juju switch-user" command.
You must first log out before logging in as another
user.

### CODE REVIEW

File: cmd/juju/user/login_test.go [hunk 1]
```diff
@@ -82,30 +82,12 @@ func (s *LoginCommandSuite) TestInit(c *gc.C) {
...
-}
-
 func (s *LoginCommandSuite) TestLogin(c *gc.C) {
-	context, args, err := s.run(c)
+	context, args, err := s.run(c, "current-user\nsekrit\nsekrit\n")
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(coretesting.Stdout(context), gc.Equals, "")
 	c.Assert(coretesting.Stderr(context), gc.Equals, `
-password: 
+username: password:
```
Comment (howbazaar) [inline cmd/juju/user/login_test.go]: Curious as to why "username:" is on the same line as "password:" but not the "type password again:".

Why is this?

File: cmd/juju/user/logout.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		}
+		loggedInCount++
+	}
+	switch loggedInCount {
+	case 0:
+		ctx.Infof("Logged out. You are no longer logged into any controllers.")
+	case 1:
+		ctx.Infof("Logged out. You are still logged into 1 controller.")
+	default:
+		ctx.Infof("Logged out. You are still logged into %d controllers.", loggedInCount)
```
Comment (howbazaar) [inline cmd/juju/user/logout.go]: Nice.

**Comment by howbazaar** [comment]
Generally LGTM, just one question about the output shown n the tests relating to new lines.
I'm assuming that this has been tested interactively?

File: cmd/juju/user/login_test.go (hunk 1 above)
Comment (axw) [inline cmd/juju/user/login_test.go]: When we read the username, output is echoed to the terminal; so when the user hits enter, a newline is printed. There's no echo for the password, so we print one out.

**Comment by axw** [comment]
> Generally LGTM, just one question about the output shown n the tests relating to new lines.
I'm assuming that this has been tested interactively?

Sure has. I had the newline after "username: " in the test originally, and then tested it interactively and took it out :)  As mentioned, because we echo the username input, there's no need for us to add a newline to the output.

**Comment by axw** [comment]
$$merge$$

**Pull request merged**